src/utils/util.js

```diff
@@ -0,0 +1,20 @@
+
+const utils = {
+  isEmail(s) {
+    /* eslint no-useless-escape:0 */
+    const regexp = /^[-a-z0-9~!$%^&*_=+}{\'?]+(\.[-a-z0-9~!$%^&*_=+}{\'?]+)*@([a-z0-9_][-a-z0-9_]*(\.[-a-z0-9_]+)*\.(aero|arpa|biz|com|coop|edu|gov|info|int|mil|museum|name|net|org|pro|travel|mobi|[a-z][a-z])|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i;
+    return regexp.test(s);
+  },
+
+  isPhone(s) {
+    /* eslint no-useless-escape:0 */
+    const regexp = /^[0-9\-\+\ ]{16}$/i;
```

Comment:
Bude validace fungovat i pro string `  420+111+222 -  `? Jak by šlo jednoduše (automaticky) zajistit kontrolu vsupu a výstupu?

Reply:
nebude, validaci jsem opravil, puivodni v zadani byla spatne, nyni ovekava +999 999 999 999

Reply:
nerozumim dotazu: "Jak by šlo jednoduše (automaticky) zajistit kontrolu vsupu a výstupu?"  kontrola obeho tam je

Reply:
@edgar0011 Takto: fce `isPhone` vrátí pro string `--420+111+222+-+` hodnotu `true` i když to není validní telefonní číslo -> to je špatně. (u čísla výše jsou mezery navíc, které ale nejsou zřetelné)  Mě by zajímalo, jak zajistíš, že se chyba nebude opakovat, až ji opravíš.

Reply:
puvodni REGEX v zadani byl tento: /^\d{9}$/i;, coz je v rozporu pak s palceholderem, nahradil jsem tim nejjednodsuim po ruce, nevim zda je cilem zadani opravovat i regexp nebo spise ukazaka prace s React a Redux, code styling a structure. Pokud je to jinak prosim o upresneni, dekuji.

Reply:
Ok, zeptám se ještě jinak :) Kak by šlo kontrolovat (nejlépe automaticky), že ti fce vrátí `false` na string `123456789`?

Reply:
nerozumim, automaticky? prece je to validacni fce ta se muze volat po submitu, behem psani/input eventy na input fieldu atd, takze by mela reagovat jen an vstup uzivatele...
a momentalne na string '123456789'  vraci false